p2p/host/eventbus/basic_metrics.go

```diff
@@ -0,0 +1,121 @@
+package eventbus
+
+import (
+	"reflect"
+	"strings"
+	"sync"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+const metricNamespace = "libp2p_eventbus"
+
+var (
+	eventsEmitted = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Namespace: metricNamespace,
+			Name:      "events_emitted_total",
+			Help:      "Events Emitted",
+		},
+		[]string{"event"},
+	)
+	totalSubscribers = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: metricNamespace,
+			Name:      "subscribers_total",
+			Help:      "Number of subscribers for an event type",
+		},
+		[]string{"event"},
+	)
+	subscriberQueueLength = prometheus.NewHistogramVec(
```

Comment:
Why is this a histogram? I think this should be a gauge. What do you think was the value in having this be a histogram?

If it's a gauge, the promql query also becomes simpler and correct. I think the current query in the grafana dashboard is wrong. It's showing you the increase in queue length. If you're full, you'll see that number go to 0 since it's not increasing.

Reply:
I wanted to look at things like 90%ile queue length. I guess we can ignore that since if the consumer is slow the queue will get full eventually.  Having a gauge alerts us to increasing queue length which should be enough. 

I didn't realise the current query would go to 0, I thought sum / count would stay constant. I just checked, it does go to 0 😅